sync-core/src/test/java/com/cloudant/sync/query/QueryExecutorTest.java

```diff
@@ -0,0 +1,351 @@
+            ds.createDocumentFromRevision(rev);
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail("Failed to create document revision");
+        }
+
+        rev.docId = "mike72";
+        bodyMap.clear();
+        bodyMap.put("name", "mike");
+        bodyMap.put("age", 72);
+        bodyMap.put("pet", "cat");
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        try {
+            ds.createDocumentFromRevision(rev);
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail("Failed to create document revision");
+        }
+
+        rev.docId = "fred34";
+        bodyMap.clear();
+        bodyMap.put("name", "fred");
+        bodyMap.put("age", 34);
+        bodyMap.put("pet", "cat");
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        try {
+            ds.createDocumentFromRevision(rev);
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail("Failed to create document revision");
+        }
+
+        rev.docId = "fred12";
+        bodyMap.clear();
+        bodyMap.put("name", "fred");
+        bodyMap.put("age", 12);
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        try {
+            ds.createDocumentFromRevision(rev);
+        } catch (IOException e) {
+            e.printStackTrace();
+            Assert.fail("Failed to create document revision");
+        }
+
+        ArrayList<Object> fieldNames = new ArrayList<Object>();
+        fieldNames.add("name");
+        fieldNames.add("age");
+        Assert.assertEquals("basic", im.ensureIndexed(fieldNames, "basic"));
+
+        fieldNames.clear();
+        fieldNames.add("name");
+        fieldNames.add("pet");
+        Assert.assertEquals("pet", im.ensureIndexed(fieldNames, "pet"));
+    }
+
+    @Test
+    public void returnsNullForNoQuery() {
+        Assert.assertNull(im.find(null));
+    }
+
+    @Test
+    public void canQueryOverOneStringField() {
+        // query - { "name" : "mike" }
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("name", "mike");
+        QueryResult queryResult = im.find(query);
+        Assert.assertNotNull(queryResult);
+        Assert.assertEquals(3, queryResult.size());
+        List<String> checkList = new ArrayList<String>();
+        checkList.add("mike12");
+        checkList.add("mike34");
+        checkList.add("mike72");
+        Assert.assertEquals(3, queryResult.documentIds().size());
+        Assert.assertTrue(queryResult.documentIds().containsAll(checkList));
+        List<String> docCheckList = new ArrayList<String>();
+        for (DocumentRevision rev: queryResult) {
+            Assert.assertNotNull(rev.getId());
+            Assert.assertNotNull(rev.getBody());
+            docCheckList.add(rev.getId());
+        }
+        Assert.assertEquals(3, docCheckList.size());
+        Assert.assertTrue(docCheckList.containsAll(checkList));
```

Comment:
Added a test to validate the iterator against documentIds and removed same validation from all other tests.  Done in 186736e.